File: src/main/java/javafiles/dataaccessfiles/FileIOWriter.java
package javafiles.dataaccessfiles;

import javafiles.Key;
import javafiles.customexceptions.ReadWriteException;

import java.util.List;
import java.util.Map;

public interface FileIOWriter {
    void writeInventory(List<Map<Key, Object>> maps) throws ReadWriteException;
}

File: src/main/java/javafiles/dataaccessfiles/fileioimplements/FileIOWriter.kt
package javafiles.dataaccessfiles.fileioimplements

import javafiles.Key
import javafiles.customexceptions.ReadWriteException
import javafiles.dataaccessfiles.FileIO

interface FileIOWriter : FileIO {
    fun writeInventory(maps: Map<Map<Key, Any>, List<Map<Key, Any>>>)
}
